### PR TITLE
MainNav - RPC Node Selector

### PR DESCRIPTION
**Pull request by candidatek**
Screenshot 2022-03-02 at 1 17 15 AM

### CODE REVIEW

File: src/context/settings.tsx [hunk 1]
```diff
@@ -6,28 +6,33 @@ import { useLocalStorageState } from '../utils'
...
 export const ENDPOINTS: IEndpoint[] = [
   {
     chainId: ENV.MainnetBeta,
+    name: "Genesysgo",
     endpoint: 'https://ssc-dao.genesysgo.net',
     network: WalletAdapterNetwork.Mainnet
   },
   {
     chainId: ENV.Devnet,
+    name: "Quick Note",
```
Comment (walt-1) [inline src/context/settings.tsx]: Should be "QuickNode"

File: src/context/settings.tsx [hunk 2]
```diff
@@ -6,28 +6,33 @@ import { useLocalStorageState } from '../utils'
...
-  endpoint: string
+  endpoint: string,
+  name: string,
   network: WalletAdapterNetwork
 }
 
 export const ENDPOINTS: IEndpoint[] = [
   {
     chainId: ENV.MainnetBeta,
+    name: "Genesysgo",
```
Comment (walt-1) [inline src/context/settings.tsx]: Should be "GenesysGo"

File: src/context/settings.tsx [hunk 3]
```diff
@@ -6,28 +6,33 @@ import { useLocalStorageState } from '../utils'
...
   },
   {
     chainId: ENV.MainnetBeta,
+    name: "Project Serum",
     endpoint: 'https://solana-api.projectserum.com',
     network: WalletAdapterNetwork.Mainnet
   },
   {
     chainId: ENV.MainnetBeta,
+    name: "Quick Note Pro",
```
Comment (walt-1) [inline src/context/settings.tsx]: Should be "QuickNode Pro"

File: src/layouts/App/SelectRPCNode.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            <span>{`${point.name}  (${point.network})`}</span>
+          </ITEM>
+        )
+      })}
+    </Menu>
+  )
+}
+
+export const SelectRPCNode: FC = () => {
+  const context = useConnectionConfig()
```
Comment (walt-1) [inline src/layouts/App/SelectRPCNode.js]: The common pattern we are trying use across the app is the ES6 object de-structuring:
`const { endpoint, endpointName, setEndpoint } = useConnectionConfig()`

File: src/layouts/App/SelectRPCNode.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import React, { FC, useEffect, useCallback, useState, Dispatch, SetStateAction } from 'react'
```
Comment (walt-1) [inline src/layouts/App/SelectRPCNode.js]: This file should be a `.tsx` file. Also could you move it to the `components` dir?

**Comment by dovahcrow** [comment]
Hi @skshriharik, may I ask if we can add an additional option for the user to input any URL for the rpc node? E.g. something like this for drift.trade.

CleanShot 2022-03-04 at 12 00 29@2x

**Comment by candidatek** [comment]
@dovahcrow Yes we can do that , but I think that would require a new design

**Comment by dovahcrow** [comment]
> @dovahcrow Yes we can do that , but I think that would require a new design

Kk got it. Then we can stick with the current one for now until later we have a new design for the custom RPC url

**Pull request merged**